packages/cardano-services/README.md

```diff
@@ -1,7 +1,92 @@
 # Cardano JS SDK | Cardano GraphQL Services
+Libraries and program entrypoints for services that facilitate remote data and submit access using 
+[Provider] interfaces over HTTP, with _optional_ queue-based transaction submission; The 
+[TxSubmitHttpService] can be configured to submit directly via [Ogmios], or via a RabbitMQ broker,
+with one or more workers handling submission and response job creation.
+
+## Features
+- [CLI] or run scripts for the [HTTP server](./src/run.ts) and [worker](./src/startWorker.ts) with 
+configuration via environment variables and _optional_ loading of secrets from disk.
+- Service port discovery via DNS resolution, or static configuration.
+- Fault-tolerant transaction submission via persistent queue, or direct submission.
+- _Optional_ [Prometheus] metrics available at `/metrics`
+
+## Services
+### HTTP Server
+The HTTP server can be started with one or more provider modules by name, segmented by URL path. 
+Run the [CLI] with `start-server --help` to see the full list of options, or inspect the 
+environment variables within [./src/run.ts](./src/run.ts). 
+
+### Worker
+A worker must be started when opting for queue-based transaction submission.
+Run the [CLI] with `start-worker --help` to see the full list of options, or inspect the
+environment variables within [./src/startWorker.ts](./src/startWorker.ts).
+
+
+## Examples
+
+_The following examples require the [install and build] steps 
+to be completed._
+
+#### All Providers | Static Service Config | Direct Tx Submission
+
+- The server will expose all [Provider] HTTP services
+- Transactions will be submitted directly via [Ogmios], running at `ws://localhost:1338`.
+- Connects to [PostgreSQL] service running at `localhost:5432`
+- HTTP API exposed using a custom API URL
+
+``` console
+./dist/cli.js \
+  start-server \
+    --api-url http://localhost:6000 \
+    --cardano-node-config-path ./config/cardano-node/config.json \
+    --ogmios-url  ws://localhost:1338 \
+    --db-connection-string postgresql://somePgUser:somePassword@localhost:5432/someDbName \
+    asset,chain-history,stake-pool,tx-submit,network-info,utxo
+```
```

Comment:
Note that CLI args are not very automation friendly. I found that `cli.js` doesn't handle environment variables quite as good as `run.js`/`startWorker.js` do.

Reply:
> Note that CLI args are not very automation friendly

Right, the idea of these docs was to demonstrate the program concept, which the CLI offers a slightly cleaner presentation IMO, and there's also `--help` to call. It's a fine balance between too little documentation and too much, the latter of which becomes a maintenance nightmare without some type of docs generation that can be automated :thinking:. That's why we've tended to just point to the Docker Compose file, integration tests, or instruct to run the CLI help, but it's time to meet in the middle with the API stability close.

Reply:
I think CLI should support environment variables and we should document both. Then user (i.e. me) don't have to guess which .js file to run and which env variables to pass if I want to rely on them.

Reply:
We'll be consolidating the two entrypoints in the next sprint. There was a technical limitation with the CLI library, but that's been removed with the introduction of a new feature :tada: